stdlib/public/core/ManagedBuffer.swift

```diff
@@ -34,22 +34,14 @@ open class ManagedBuffer<Header, Element> {
       ManagedBuffer<Header, Element>) throws -> Header
   ) rethrows -> ManagedBuffer<Header, Element> {
 
-    let p = try ManagedBufferPointer<Header, Element>(
-      bufferClass: self,
-      minimumCapacity: minimumCapacity,
-      makingHeaderWith: { buffer, _ in
-        try factory(
-          unsafeDowncast(buffer, to: ManagedBuffer<Header, Element>.self))
-      })
-
-    return unsafeDowncast(p.buffer, to: ManagedBuffer<Header, Element>.self)
-  }
+    let p = Builtin.allocWithTailElems_1(
+         self,
+         minimumCapacity._builtinWordValue, Element.self)
 
-  /// Destroy the stored Header.
-  deinit {
-    ManagedBufferPointer(self).withUnsafeMutablePointerToHeader {
-      _ = $0.deinitialize()
-    }
+    let initHeaderVal = try factory(p)
+    p.headerAddress.initialize(to: initHeaderVal)
+    _fixLifetime(p) 
```

Comment:
Might be worth checking with @rjmccall and @gottesmm to see if this doesn't already fall under the derived-pointer rules for ARC. If `headerAddress` is a stored property or else `@_transparent` it may not be necessary to explicitly fix the lifetime anymore.